src/reconcile/rec_write.c

```diff
@@ -433,10 +446,18 @@ __wt_reconcile(WT_SESSION_IMPL *session, WT_REF *ref,
 	WT_ILLEGAL_VALUE_SET(session);
 	}
 
-	/* Checks for a successful reconciliation. */
+	/* Check for a successful reconciliation. */
 	if (ret == 0)
 		ret = __rec_write_check_complete(session, r, lookaside_retryp);
 
+	/*
+	 * XXX special case to fall back to lookaside eviction during
+	 * checkpoints if a page can't be evicted.
+	 */
+	if (ret == EBUSY && lookaside_retryp != NULL &&
+	    !F_ISSET(r, WT_REC_UPDATE_RESTORE) && !r->update_uncommitted)
+		*lookaside_retryp = true;
+
```

Comment:
Is the `lookaside_retryp` handling split between `__wt_reconcile()` and `__rec_write_check_complete()` because it makes it easier to catch the `EBUSY` case? I'd suggest moving all of the code into `__rec_write_check_complete()`, but I wouldn't argue about it.